qemu/tests/timerdevice_clock_drift_with_sleep.py

```diff
@@ -98,36 +99,37 @@ def verify_elapsed_time():
         vm.verify_alive()
         session = vm.wait_for_login(timeout=timeout)
 
-    error.context("Stop auto sync service in guest", logging.info)
+    error_context.context("Stop auto sync service in guest", logging.info)
     cmd = "(service chronyd status | grep 'Loaded: loaded')"
     cmd += " && service chronyd stop"
     session.cmd_status_output(cmd)
 
-    error.context("Sync time from guest to ntpserver", logging.info)
-    session.cmd("yum install -y ntpdate; ntpdate clock.redhat.com",
-                timeout=timeout)
+    error_context.context("Sync time from guest to ntpserver", logging.info)
 
-    error.context("Sleep a while and check the time drift on guest"
-                  " (without any pinned vcpu)", logging.info)
+    session.cmd(ntp_sync_cmd, timeout=timeout)
+
+    error_context.context("Sleep a while and check the time drift on guest"
+                          " (without any pinned vcpu)", logging.info)
     verify_elapsed_time()
 
-    error.context("Pin every vcpu to physical cpu", logging.info)
+    error_context.context("Pin every vcpu to physical cpu", logging.info)
     host_cpu_cnt_cmd = params["host_cpu_cnt_cmd"]
-    host_cpu_num = utils.system_output(host_cpu_cnt_cmd).strip()
+    host_cpu_num = process.system_output(host_cpu_cnt_cmd, shell=True).strip()
     host_cpu_list = (_ for _ in range(int(host_cpu_num)))
     cpu_pin_list = zip(vm.vcpu_threads, host_cpu_list)
     if len(cpu_pin_list) < len(vm.vcpu_threads):
-        raise error.TestNAError("There isn't enough physical cpu to"
-                                " pin all the vcpus")
+        test.error("There isn't enough physical cpu to"
```

Comment:
should this be `test.cancel()`

Reply:
Agree, and if so maybe better move the judgement to the beginning of the test case .